examples/question_answer/README.md

```diff
@@ -0,0 +1,144 @@
+end `/pairing`
+
+**NB**: This will place the above mentioned properties in the ESP-IDF sdk project configuration.
+Cleaning the configuration will also reset them to their default values. If you want to store them
+more permanently you can overwrite the defaults by adding them in the `sdkconfig.defaults` file.
+
+## Building and flashing
+
+Once the project has been configured build and flash it using the ESP-IDF utilities.
+
+You can build, flash and then open a monitor using:
+| ESP-IDF v4.x  | ESP-IDF v3.x |
+| ------------- | ------------ |
+| `idf.py -p <DEVICE_PORT> build flash monitor` | `make ESPPORT=<DEVICE_PORT> flash monitor` |
+
+When re-using an old device that might have been information stored in persistent flash,
+clear the flash. From terminal use the command:
+| ESP-IDF v4.x  | ESP-IDF v3.x |
+| ------------- | ------------ |
+| `idf.py -p <DEVICE_PORT> erase-flash` | `make ESPPORT=<DEVICE_PORT> erase_flash` |
+
+## Device connection to Astarte
+
+Once the example has been flashed it will automatically attempt to connect the device to the remote
+instance of Astarte.
+Verify this by using the monitor functionality of the ESP-IDF to observe the log messages of the
+ESP32.
+After connection has been accomplished, the following ESP32 log message should be shown:
+```
+ASTARTE_QUESTION_ANSWER: Astarte device connected, session_present: 0
+```
+You can see how the connection has been achieved also by from the Astarte Dashboard by opening the
+newly created device and observing the status that should have changed to *Connected*.
+
+## Send and receive data from the device
+
+Now that our device is connected to Astarte is time to start sending and receiving messages from it.
+We will do that through the `astartectl` tool.
+
+`astartectl` needs an access token to interact with the remote Astarte instance. We can find this
+token in the *Astarte Cloud Console* (the page on Astarte Cloud where all your realms are shown) in
+the *Realm Information* page.
+
+The token is presented as a command that will store it in a shell variable. Something similar to:
+`TOKEN="token_content"`.
+
+We will use two useful commands in this example, `send-data` and `get-samples`.
+With `send-data` we can send our question to the device. The syntax is the following:
+```
+astartectl --astarte-url "<API_URL>" --token "<TOKEN>" appengine --realm-name "<REALM_NAME>" \
+    devices send-data <DEVICE_NAME>                                                          \
+    org.astarteplatform.esp32.ServerDatastream /question <VALUE>
+```
+Where `<TOKEN>` is the shell variable containing the token, `<REALM_NAME>` is your realm's name and
+`<API_URL>` is the API's url as found on astarte cloud console. `<DEVICE_NAME>` is the device ID to
+send the data to, and `<VALUE>` is the value to send. In this example the data to send is in boolean format, so substitute `<VALUE>` with `true` or
```

Comment:
This line is way longer than the others. Is it wanted?

Reply:
Nope, no clue how it slipped in.